tests/php/Service/RoomServiceTest.php

```diff
@@ -0,0 +1,253 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2020 Joas Schilling <coding@schilljs.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Tests\php\Service;
+
+use OCA\Talk\Exceptions\RoomNotFoundException;
+use OCA\Talk\Manager;
+use OCA\Talk\Participant;
+use OCA\Talk\Room;
+use OCA\Talk\Service\RoomService;
+use OCP\IUser;
+use PHPUnit\Framework\MockObject\MockObject;
+use Test\TestCase;
+
+class RoomServiceTest extends TestCase {
+
+	/** @var Manager|MockObject */
+	protected $manager;
+	/** @var RoomService */
+	private $service;
+
+
+	public function setUp(): void {
+		parent::setUp();
+
+		$this->manager = $this->createMock(Manager::class);
+		$this->service = new RoomService(
+			$this->manager
+		);
+	}
+
+	public function testCreateOneToOneConversationWithSameUser(): void {
+		$user = $this->createMock(IUser::class);
+		$user->expects($this->exactly(2))
+			->method('getUID')
+			->willReturn('uid');
```

Comment:
Isn't the fact how many times `getUID` is called an implementation detail that doesn't affect the result?

```suggestion
		$user->method('getUID')
			->willReturn('uid');
```

Same for all usages below.

Reply:
done

Reply:
All except this one :smile:

Reply:
Ops